src/components/User/UserComponent.jsx

```diff
@@ -1,120 +1,253 @@
+      this.props.loading(true);
+      return fetchData(url).then((response) => {
+        this.props.loading(false);
+        this.setState({ users: response.data.results });
+      }).catch(() => {
+        this.props.loading(false);
+        this.setState({ allDataFetched: true });
+      });
+    }
+    return this.props.loadUsers(activePage, limit);
+  };
+
+  handlePageTotal = () => Math.ceil(this.props.usersCount / this.state.limit);
+
+  emptyUsersList = () => (isEmpty(this.props.users));
+
+  filterApplied = () => {
+    this.setState({ isFiltered: !this.state.isFiltered });
   }
 
-  if (hasNoUsers) {
-    return (
-      <NavBarComponent>
-        <ItemsNotFoundComponent allDataFetched={props.allDataFetched} message="Please try again later, to see if we'll have users to show you." />
-      </NavBarComponent>
+  render() {
+    const currentUsers = `page_${this.props.activePage}`;
+    const activePageUsers = this.props.users[currentUsers] || this.state.users;
+    const hasNoUsers = isEmpty(activePageUsers);
+    const showStatus = this.props.hasError && this.props.errorMessage;
+
+    const filterButton = () => (
+      <FilterButton
+        activePage={this.props.activePage}
+        limit={this.state.limit}
+        selected={this.props.selected}
+        filterAction={this.props.loadUsers}
+        disabled={this.props.isLoading}
+        option={this.props.filterData[1]}
+        filterApplied={this.filterApplied}
+      >
+        <React.Fragment>
+          <FilterComponent
+            index={0}
+            option={this.props.filterData[0]}
+            selected={this.props.selected}
+            filterSelection={this.props.filterSelection}
+          />
+
+          <FilterComponent
+            index={1}
+            option={this.props.filterData[1]}
+            selected={this.props.selected}
+            filterSelection={this.props.filterSelection}
+          />
+        </React.Fragment>
+      </FilterButton>
     );
-  }
 
-  return (
-    <div>
-      <Table basic selectable>
-        <Table.Header>
-          <Table.Row>
-            <Table.HeaderCell>Name</Table.HeaderCell>
-            <Table.HeaderCell>Email Address</Table.HeaderCell>
-            <Table.HeaderCell>Cohort</Table.HeaderCell>
-            <Table.HeaderCell>Assets Assigned</Table.HeaderCell>
-          </Table.Row>
-        </Table.Header>
-        <Table.Body>
-          {props.activePageUsers.map((user) => {
-            const viewUserUrl = `users/${user.id}/view`;
-
-            const updatedUser = {
-              ...user,
-              assets_assigned: user.allocated_asset_count
-            };
-
-            return (
-              <TableRow
-                viewDetailsRoute={viewUserUrl}
-                key={user.id}
-                data={updatedUser}
-                headings={['full_name', 'email', 'cohort', 'assets_assigned']}
+    if (this.props.isLoading) {
+      return <LoaderComponent />;
+    }
+
+    if (hasNoUsers) {
```

Comment:
This section can be made shorted by creating a header component as illustrated [here](https://gist.github.com/sachungo/fec45438336a94dcf5884d81c389c024)